board.cpp

```diff
@@ -0,0 +1,164 @@
+#include "board.h"
+
+bool isPositionValid(Board board, Position pos){ //check if pos if withing acceptable values 
+    if(pos.xpos < 0 || pos.ypos < 0) return false;
+    if(pos.xpos > board.width || pos.ypos > board.height) return false;

```

Comment:
#should avoid braceless conditional and looping constructs.

I'll leave up to you and your style to honor this comment or not, but I strongly suggest not to omit braces on these places - it's super error-prone.